src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusSerializerExt.cs

```diff
@@ -0,0 +1,206 @@
+                    var values = metricPoint.Values;
+                    var timestamp = metricPoint.EndTime.ToUnixTimeMilliseconds();
+
+                    // Counter and Gauge
+                    cursor = WriteUnicodeStringNoEscape(buffer, cursor, metric.Name);
+                    buffer[cursor++] = unchecked((byte)' ');
+                    buffer[cursor++] = unchecked((byte)'{');
+
+                    for (var i = 0; i < keys.Length; i++)
+                    {
+                        if (i > 0)
+                        {
+                            buffer[cursor++] = unchecked((byte)',');
+                        }
+
+                        cursor = WriteLabel(buffer, cursor, keys[i], values[i]);
+                    }
+
+                    buffer[cursor++] = unchecked((byte)'}');
+                    buffer[cursor++] = unchecked((byte)' ');
+
+                    if (((int)metric.MetricType & 0b_0000_1111) == 0x0a /* I8 */)
+                    {
+                        cursor = WriteLong(buffer, cursor, metricPoint.LongValue);
+                    }
+                    else
+                    {
+                        cursor = WriteDouble(buffer, cursor, metricPoint.DoubleValue);
+                    }
+
+                    buffer[cursor++] = unchecked((byte)' ');
+                    cursor = WriteLong(buffer, cursor, timestamp);
+
+                    buffer[cursor++] = ASCII_LINEFEED;
+                }
+            }
+            else
+            {
+                foreach (ref var metricPoint in metric.GetMetricPoints())
+                {
+                    var keys = metricPoint.Keys;
+                    var values = metricPoint.Values;
+                    var timestamp = metricPoint.EndTime.ToUnixTimeMilliseconds();
+
+                    // Histogram sum
+                    cursor = WriteUnicodeStringNoEscape(buffer, cursor, metric.Name);
+                    buffer[cursor++] = unchecked((byte)'_');
+                    buffer[cursor++] = unchecked((byte)'s');
+                    buffer[cursor++] = unchecked((byte)'u');
+                    buffer[cursor++] = unchecked((byte)'m');
+                    buffer[cursor++] = unchecked((byte)' ');
+                    buffer[cursor++] = unchecked((byte)'{');
+
+                    for (var i = 0; i < keys.Length; i++)
+                    {
+                        if (i > 0)
+                        {
+                            buffer[cursor++] = unchecked((byte)',');
+                        }
+
+                        cursor = WriteLabel(buffer, cursor, keys[i], values[i]);
+                    }
+
+                    buffer[cursor++] = unchecked((byte)'}');
+                    buffer[cursor++] = unchecked((byte)' ');
+
+                    cursor = WriteDouble(buffer, cursor, metricPoint.DoubleValue);
+
+                    buffer[cursor++] = unchecked((byte)' ');
+                    cursor = WriteLong(buffer, cursor, timestamp);
```

Comment:
Instead of doing serialization each time, perhaps it'll save some CPU cycles if this is serialized outside the loop, and reused while in the loop.